Framework/Core/src/AODReaderHelpers.cxx

```diff
@@ -185,25 +185,25 @@ AlgorithmSpec AODReaderHelpers::rootFileReaderCallback()
     }
 
     // get the run time watchdog
-    auto* watchdog = new RuntimeWatchdog(options.get<int64_t>("time-limit"));
+    auto watchdog = std::make_shared<RuntimeWatchdog>(options.get<int64_t>("time-limit"));
 
     // selected the TFN input and
     // create list of requested tables
-    header::DataHeader TFNumberHeader;
-    std::vector<OutputRoute> requestedTables;
+    auto tfNumberHeader = std::make_shared<header::DataHeader>();
+    auto requestedTables = std::make_shared<std::vector<OutputRoute>>();
```

Comment:
Why does it need to be a shared_ptr?

Reply:
I was trying to find out where the seg fault is coming from you pointed out yesterday. It seems that this happens when the reader is closed by the data processing and I tested whether this is related with these variables of the reader. But it is not. So shared_ptr is not needed and can be reverted.

Reply:
Ok, thanks. Please provide the minimal fix for the issue mentioned in the title so that we can better track changes.